### DISCUSSION TITLE
selectRowList как собать все данные с листа

### DISCUSSION THREAD

**Discussion by Murbulat1** [totumonline/totum-ru-issues-and-discussions#393] [Как это сделать?] [answered]
Добрый день, пытаюсь разобратся с данными. Собираю все данные с расчетного цикла с помощью selectRowList, все работает. Можно ли туда же прикрутить чтобы он еще собирал с полей из header и footer. Пытался использовать sfield / sfields так и не понял как они работают.

Если создать selectRow по хедеру или футеру то он там собирает, опять же игнорируя сам контейнер листа

**Comment by erpic-ru** [comment, marked as answer]
Там 2 типа данных: список массивов и массив, поэтому оно не очень объединяется.

Сделайте 2-3 отдельных запроса и объедините, при необходимости, так как вам нужно.

Также можно передать строковое значение из списка в виде fields: "*ALL*"  когда нужны все, чтобы не прописывать все поля отдельно:

    *ALL* — все поля строчной части.
    *HEADER* — все поля хедера.
    *FOOTER* — все поля футера.

sfield берет отображение полей селектов, а не значение, sfields - собственно, список полей, где нужно взять отображение

**Reply by Murbulat1** [reply]
А обьеденить их с помощью str, и потом уже вывести его из цикла ?

**Reply by Murbulat1** [reply]
Пока что нашел решение сделать через **listReplace**. Где я просто задаю поля из хедера либо футера, в list указал selectRowList В принципе работает, не знаю как повлияет на производительность

**Reply by ttm-alexeypolunin** [reply]
```
=: rowCreate(field: "header" = $header; field: "rows" = $rows; field: "footer" = $footer)

header: selectRow(table: $#ntn; fields: "*HEADER*")
rows: selectRowList(table: $#ntn; fields: "*ALL*")
footer: selectRow(table: $#ntn; fields: "*FOOTER*")
```
